File: ibkr_dash_frontend/src/types/dividends.ts
import type { PaginationInfo } from './common'

export interface DividendItem {
  account_id: string
  currency: string | null
  symbol: string | null
  description: string | null
  date_time: string | null
  settle_date: string | null
  amount: number | null
  flow_type: string | null
  dividend_type: string | null
  transaction_id: string | null
  report_date: string | null
  ex_date: string | null
}

export interface DividendListResponse {
  items: DividendItem[]
  pagination: PaginationInfo
}

export interface DividendSummaryResponse {
  record_count: number
  dividend_count: number
  withholding_tax_count: number
  gross_dividend_amount: number | null
  withholding_tax_amount: number | null
  net_amount: number | null
  by_currency: DividendCurrencySummaryItem[]
}

export interface DividendCurrencySummaryItem {
  currency: string | null
  record_count: number
  dividend_count: number
  withholding_tax_count: number
  gross_dividend_amount: number
  withholding_tax_amount: number
  net_amount: number
}

File: backend/app/schemas/dividends.py
"""Dividend schemas (cash flows filtered to dividend-related flow types)."""

from pydantic import BaseModel

from app.schemas.common import PaginationInfo


class DividendItem(BaseModel):
    """A single dividend or withholding-tax cash flow record."""
    account_id: str
    currency: str | None = None
    symbol: str | None = None
    description: str | None = None
    date_time: str | None = None
    settle_date: str | None = None
    amount: float | None = None
    flow_type: str | None = None
    dividend_type: str | None = None
    transaction_id: str | None = None


class DividendListResponse(BaseModel):
    """Paginated dividend list."""
    items: list[DividendItem]
    pagination: PaginationInfo


class DividendCurrencySummaryItem(BaseModel):
    """Summary for a single currency."""
    currency: str | None = None
    record_count: int = 0
    dividend_count: int = 0
    withholding_tax_count: int = 0
    gross_dividend_amount: float = 0.0
    withholding_tax_amount: float = 0.0
    net_amount: float = 0.0


class DividendSummaryResponse(BaseModel):
    """Aggregated dividend summary."""
    record_count: int = 0
    dividend_count: int = 0
    withholding_tax_count: int = 0
    gross_dividend_amount: float | None = None
    withholding_tax_amount: float | None = None
    net_amount: float | None = None
    by_currency: list[DividendCurrencySummaryItem] = []
